express-api/src/services/properties/propertiesServices.ts

```diff
@@ -104,6 +104,11 @@ const collectFindOptions = (filter: PropertyUnionFilter) => {
     options.push(constructFindOptionFromQuery('AdministrativeArea', filter.administrativeArea));
   if (filter.propertyType)
     options.push(constructFindOptionFromQuery('PropertyType', filter.propertyType));
+  if (filter.agencyId && filter.agencyId.length > 0) {
+    filter.agencyId.forEach((id) => {
+      options.push(constructFindOptionFromQuery('AgencyId', `in,${String(id)}`));
+    });
+  }
```

Comment:
I think doing it this way will add to the OR query. It really needs to be combined in a way that it is a requirement, not an option. Potentially like this, someone could search by agency name and get those properties even if they aren't in their own agencies, right?